packages/rocketchat-ui-sidenav/client/chatRoomItem.js

```diff
@@ -34,18 +34,27 @@ Template.chatRoomItem.helpers({
 			unread,
 			active,
 			archivedClass,
-			statusClass: this.t === 'd' ? Session.get(`user_${ this.name }_status`) || 'offline' : this.t === 'l' ? RocketChat.roomTypes.getUserStatus(this.t, this.rid) || 'offline' : false
+			status: this.t === 'd' || this.t === 'l'
 		};
 
-		if (RocketChat.settings.get('Store_Last_Message')) {
-			if (this.lastMessage) {
-				roomData.lastMessage = this.lastMessage;
-			} else {
-				const room = RocketChat.models.Rooms.findOne(this.rid || this._id, { fields: { lastMessage: 1 } });
-				roomData.lastMessage = room && room.lastMessage || { msg: t('No_messages_yet') };
-			}
+		if (!this.lastMessage && RocketChat.settings.get('Store_Last_Message')) {
+			const room = RocketChat.models.Rooms.findOne(this.rid || this._id, { fields: { lastMessage: 1 } });
+			roomData.lastMessage = room && room.lastMessage || { msg: t('No_messages_yet') };
 		}
-
 		return roomData;
 	}
 });
+
+RocketChat.callbacks.add('enter-room', (sub) => {
+	const items = $('.rooms-list .sidebar-item');
+	items.filter('.sidebar-item--active').removeClass('sidebar-item--active');
+	items.filter(`[data-id=${ sub.rid }]`).addClass('sidebar-item--active');
+	return sub;
+});
+
+Template.sidebarItemStatus.helpers({
+	statusClass() {
```

Comment:
this is not working as expected.. it is showing `offline` for all users

Reply:
are you sure? 
![image](https://user-images.githubusercontent.com/5263975/37939652-65316f2c-313a-11e8-82f0-3e48f0e48fbb.png)

Reply:
did you test with any other user or just yours?